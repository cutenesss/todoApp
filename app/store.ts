import { configureStore, PreloadedState } from '@reduxjs/toolkit'
import RootReducer from './rootReducer'
import Reactotron from './config/ReactotronConfig'
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { useDispatch, TypedUseSelectorHook, useSelector } from 'react-redux'
import DebugConfig from './config/DebugConfig'

// const enhancers = DebugConfig.reactotron ? [Reactotron.createEnhancer!()] : []

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: RootReducer,
    middleware: getDefaultMiddleware => getDefaultMiddleware(),
    preloadedState
  // enhancers,
  })
}

export type RootState = ReturnType<typeof RootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
