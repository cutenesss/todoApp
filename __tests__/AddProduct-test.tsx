/**
 * @format
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../app/store';
import AddEditProductScreen from '../app/screens/AddEditProduct';

test('render add product list page', () => {
    const tree = renderer.create(<Provider store={store}>
        <AddEditProductScreen />
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})
