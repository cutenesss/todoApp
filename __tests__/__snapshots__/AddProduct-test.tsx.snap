// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render add product list page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "marginTop": 15,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Name"
      secureTextEntry={false}
      style={
        Object {
          "borderColor": "#D9D9D9",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "minHeight": 40,
          "paddingHorizontal": 10,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 15,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Descripte"
      secureTextEntry={false}
      style={
        Object {
          "borderColor": "#D9D9D9",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "minHeight": 40,
          "paddingHorizontal": 10,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 15,
        },
      ]
    }
  >
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Price"
      secureTextEntry={false}
      style={
        Object {
          "borderColor": "#D9D9D9",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "minHeight": 40,
          "paddingHorizontal": 10,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 10,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Add
    </Text>
  </View>
</View>
`;
